tests/sae/test_sae_core.py

```
@@ -0,0 +1,149 @@
+    out = core.embed(["ACDE", "ACDE"])
+    # Recompute expected pooled vector for sequence 0 (last residue is padding).
+    with torch.no_grad():
+        layer_states = torch.as_tensor(hidden[1, 0], dtype=torch.float32)
+        acts = sae.encode(layer_states)
+        valid = torch.as_tensor(chain_index[0] != -1)
+        expected = acts[valid].max(dim=0).values.numpy()
+    np.testing.assert_allclose(out.pooled_features[0], expected, rtol=1e-5, atol=1e-6)
+
+
+def test_padding_residue_excluded_from_pool() -> None:
+    core, embedder, sae, hidden, chain_index = _make_core()
```

Ruff (0.16.1)

[warning] Unpacked variable `embedder` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)